tabular/src/autogluon/tabular/models/automm/automm_model.py

```diff
@@ -0,0 +1,216 @@
+        default_ag_args.update(extra_ag_args)
+        return default_ag_args
+
+    def _set_default_params(self):
+        super()._set_default_params()
+        try_import_autogluon_text()
+
+    def _fit(self,
+             X: pd.DataFrame,
+             y: pd.Series,
+             X_val: Optional[pd.DataFrame] = None,
+             y_val: Optional[pd.Series] = None,
+             time_limit: Optional[int] = None,
+             sample_weight=None,
+             **kwargs):
+        """The internal fit function
+
+        Parameters
+        ----------
+        X
+            Features of the training dataset
+        y
+            Labels of the training dataset
+        X_val
+            Features of the validation dataset
+        y_val
+            Labels of the validation dataset
+        time_limit
+            The time limits for the fit function
+        sample_weight
+            The weights of the samples
+        kwargs
+            Other keyword arguments
+
+        """
+        try_import_autogluon_text()
+        from autogluon.text.automm import AutoMMPredictor
+
+        # Decide name of the label column
+        if 'label' in X.columns:
+            label_col_id = 0
+            while True:
+                self._label_column_name = 'label{}'.format(label_col_id)
+                if self._label_column_name not in X.columns:
+                    break
+                label_col_id += 1
+        else:
+            self._label_column_name = 'label'
+        X_train = self.preprocess(X, fit=True)
+        if X_val is not None:
+            X_val = self.preprocess(X_val)
+        # Get arguments from kwargs
+        verbosity = kwargs.get('verbosity', 2)
+        num_gpus = kwargs.get('num_gpus', None)
+        if sample_weight is not None:  # TODO: support
+            logger.log(15, "sample_weight not yet supported for AutoMMPredictorModel, "
+                           "this model will ignore them in training.")
+
+        X_train.insert(len(X_train.columns), self._label_column_name, y)
+        if X_val is not None:
+            X_val.insert(len(X_val.columns), self._label_column_name, y_val)
+
+        verbosity_text = max(0, verbosity - 1)
+        root_logger = logging.getLogger('autogluon')
+        root_log_level = root_logger.level
+        self.model = AutoMMPredictor(label=self._label_column_name,
+                                     problem_type=self.problem_type,
+                                     path=self.path,
+                                     eval_metric=self.eval_metric,
+                                     verbosity=verbosity_text)
+        params = self._get_model_params()
+
+        if num_gpus is not None:
+            params['env.num_gpus'] = num_gpus
+
+        self.model.fit(train_data=X_train,
+                       tuning_data=X_val,
+                       time_limit=time_limit,
+                       hyperparameters=params,
+                       seed=params.pop('seed', 0))
```

Comment:
Why are we popping params here? This is a bit iffy, I'm not even technically sure what happens here, but I assume `hyperparameters` will have the popped version of params. Maybe better to explicitly get the seed popped from params at an earlier point instead of inside the model.fit call for better readability.